x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -103,7 +103,15 @@ export class CloudPlugin implements Plugin<CloudSetup> {
   }
 
   public start(coreStart: CoreStart): CloudStart {
-    coreStart.chrome.setHelpSupportUrl(ELASTIC_SUPPORT_LINK);
+    let supportUrl = ELASTIC_SUPPORT_LINK;
+    if (this.config.serverless?.project_id) {
+      // serverless projects use config.id and config.serverless.project_id
+      supportUrl += '?serverless_project_id=' + this.config.serverless.project_id;
+    } else if (this.config.id) {
+      // non-serverless Cloud projects only use config.id
+      supportUrl += '?cloud_deployment_id=' + this.config.id;
+    }
```

Comment:
NIT: I would extract that to an `getSupportUrl(config: CloudConfigType): string` helper function